app/src/main/java/com/sdp13epfl2021/projmag/activities/ProjectInformationActivity.kt

```diff
@@ -394,29 +404,37 @@ class ProjectInformationActivity : AppCompatActivity() {
         }
     }
 
-    private fun handleVideoWithFavoritePersistenceFiltering(videosLinks: List<String>) {
-        //get the favorite list and when it's available provide it to the function that is
-        //responsible for downloading videos in volatile or non volatile memory
+
+    /**
+     * Function that downloads a video either in the proj dir permanent app memory if it's
+     * a favorite or one of our own projects. Else, the video is stored in the cache dir which is a
+     * memory that is deleted by the phone if it needs it. This function calls a callback and will do
+     * nothing if the userDatabase or the projectDatabase didn't respond for some reason.
```

Comment:
The documentation about the callback seems a little weird.
What do you mean by "didn't respond for some reason" ?

Reply:
I just meant if it returns onFailure() I do nothing in this case

Reply:
didn't respond and call the failure callback are not the same.
Also, this function does not call the callback, it's database that calls the callback.